### PR TITLE
refactor: attended members

### PR DESCRIPTION
**Pull request by Prerna290**
## Type
- ( ) Fix
- (X) Refactor
- ( ) Style
- ( ) Docs
- ( ) Feat
- ( ) Hotfix

### CODE REVIEW

File: apps/commudle-admin/src/app/feature-modules/listing-pages/public-home-list-events/components/events-volunteer-list/events-volunteer-list.component.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+@Component({
+  selector: 'commudle-events-volunteer-list',
+  templateUrl: './events-volunteer-list.component.html',
+  styleUrls: ['./events-volunteer-list.component.scss'],
+})
+export class EventsVolunteerListComponent implements OnInit {
+  @Input() event: IEvent;
+  volunteers: IUser[] = [];
+  page_info: IPageInfo;
```
Comment (ArshdeepGrover) [inline apps/commudle-admin/src/app/feature-modules/listing-pages/public-home-list-events/components/events-volunteer-list/events-volunteer-list.component.ts]: Let,s you camel case for variable name

File: apps/commudle-admin/src/app/feature-modules/public-events/components/attended-members/attended-members-card/attended-members-card.component.ts [hunk 2]
```diff
@@ -35,4 +36,8 @@ export class AttendedMembersCardComponent implements OnInit {
   openChatWithUser(): void {
     this.userChatsService.changeFollowerId(this.user.id);
   }
+
+  CommonEvents() {
```
Comment (ArshdeepGrover) [inline apps/commudle-admin/src/app/feature-modules/public-events/components/attended-members/attended-members-card/attended-members-card.component.ts]: Let'S use more good name for functions?

File: apps/commudle-admin/src/app/feature-modules/public-events/components/attended-members/attended-members.component.html [hunk 3]
```diff
@@ -30,13 +50,42 @@
               [current]="page"
               [total]="total"
             ></app-pagination>
-          </ng-container>
```
Comment (ArshdeepGrover) [inline apps/commudle-admin/src/app/feature-modules/public-events/components/attended-members/attended-members.component.html:33]: Why we are removing ng-container?

File: apps/commudle-admin/src/app/feature-modules/public-events/components/attended-members/attended-members.component.ts [hunk 4]
```diff
@@ -66,18 +93,67 @@ export class AttendedMembersComponent implements OnInit, OnDestroy {
 
   getMembers(): void {
     this.isLoading = true;
-    this.subscriptions.push(
```
Comment (ArshdeepGrover) [inline apps/commudle-admin/src/app/feature-modules/public-events/components/attended-members/attended-members.component.ts:69]: Why we are remving subscription?

**Review by ArshdeepGrover — CHANGES_REQUESTED** [review]

File: apps/commudle-admin/src/app/feature-modules/listing-pages/public-home-list-events/components/events-volunteer-list/events-volunteer-list.component.ts (hunk 1 above)
Comment (Prerna290) [reply apps/commudle-admin/src/app/feature-modules/listing-pages/public-home-list-events/components/events-volunteer-list/events-volunteer-list.component.ts]: Done

File: apps/commudle-admin/src/app/feature-modules/public-events/components/attended-members/attended-members-card/attended-members-card.component.ts (hunk 2 above)
Comment (Prerna290) [reply apps/commudle-admin/src/app/feature-modules/public-events/components/attended-members/attended-members-card/attended-members-card.component.ts]: Done

File: apps/commudle-admin/src/app/feature-modules/public-events/components/attended-members/attended-members-card/attended-members-card.component.ts [hunk 5]
```diff
@@ -35,4 +36,8 @@ export class AttendedMembersCardComponent implements OnInit {
   openChatWithUser(): void {
     this.userChatsService.changeFollowerId(this.user.id);
   }
+
+  CommonEventsAttended() {
```
Comment (ArshdeepGrover) [inline apps/commudle-admin/src/app/feature-modules/public-events/components/attended-members/attended-members-card/attended-members-card.component.ts]: let's rename this function name to better way

File: apps/commudle-admin/src/app/feature-modules/public-events/components/attended-members/attended-members.component.ts [hunk 6]
```diff
@@ -17,46 +20,70 @@ export class AttendedMembersComponent implements OnInit, OnDestroy {
   event: IEvent;
   members: IUser[] = [];
   isLoading = false;
-
+  employee = false;
+  employer = false;
+  page_info: IPageInfo;
```
Comment (ArshdeepGrover) [inline apps/commudle-admin/src/app/feature-modules/public-events/components/attended-members/attended-members.component.ts]: let's rename this to camel case!

File: apps/commudle-admin/src/app/feature-modules/public-events/components/attended-members/attended-members.component.ts [hunk 7]
```diff
@@ -66,18 +93,67 @@ export class AttendedMembersComponent implements OnInit, OnDestroy {
...
   getMembers(): void {
     this.isLoading = true;
-    this.subscriptions.push(
-      this.eventsService.getAttendedMembers(this.query, this.event.id, this.page, this.count).subscribe((data) => {
+    if (!this.canLoadMore) {
+      return;
+    }
+    this.canLoadMore = false;
+    this.eventsService
+      .getAttendedMembers(this.page, this.count, this.event.id, this.query, this.employer, this.employee)
```
Comment (ArshdeepGrover) [inline apps/commudle-admin/src/app/feature-modules/public-events/components/attended-members/attended-members.component.ts]: let's use subscriptions.push

File: apps/commudle-admin/src/app/feature-modules/public-events/components/team/team.component.ts [hunk 8]
```diff
@@ -28,7 +28,7 @@ export class TeamComponent implements OnInit {
 
   getVolunteers() {
     this.eventsService.pGetEventVolunteers(this.event.id).subscribe((data) => {
```
Comment (ArshdeepGrover) [inline apps/commudle-admin/src/app/feature-modules/public-events/components/team/team.component.ts]: let's use subscriptions.push

File: apps/commudle-admin/src/app/feature-modules/listing-pages/public-home-list-events/components/events-volunteer-list/events-volunteer-list.component.ts [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    this.getVolunteersList();
+  }
+
+  getVolunteersList() {
+    this.showSpinner = true;
+    if (this.isLoadingVolunteers) {
+      return;
+    }
+    this.isLoadingVolunteers = true;
+    this.eventsService.pGetEventVolunteers(this.event.id).subscribe((data) => {
```
Comment (ArshdeepGrover) [inline apps/commudle-admin/src/app/feature-modules/listing-pages/public-home-list-events/components/events-volunteer-list/events-volunteer-list.component.ts]: let's use subscriptions.push

**Review by ArshdeepGrover — CHANGES_REQUESTED** [review]

**Review by ArshdeepGrover — APPROVED** [review]

**Pull request merged**